pkg/common/gatewayapi_utils.go

```diff
@@ -447,3 +449,37 @@ func GetGatewayWorkloadSelector(ctx context.Context, cli client.Client, gateway
 	}
 	return GetServiceWorkloadSelector(ctx, cli, serviceKey)
 }
+
+func IsHTTPRouteAccepted(httpRoute *gatewayapiv1alpha2.HTTPRoute) bool {
+	if httpRoute == nil {
+		return false
+	}
+
+	if len(httpRoute.Spec.CommonRouteSpec.ParentRefs) == 0 {
+		return false
+	}
+
+	// Check HTTProute parents (gateways) in the status object
+	// if any of the current parent gateways reports not "Admitted", return false
+	for _, parentRef := range httpRoute.Spec.CommonRouteSpec.ParentRefs {
+		routeParentStatus := func(pRef gatewayapiv1alpha2.ParentReference) *gatewayapiv1alpha2.RouteParentStatus {
+			for idx := range httpRoute.Status.RouteStatus.Parents {
+				if reflect.DeepEqual(pRef, httpRoute.Status.RouteStatus.Parents[idx].ParentRef) {
+					return &httpRoute.Status.RouteStatus.Parents[idx]
```

Comment:
What about returning false if not deepEqual here? and then in line 472 return true

Reply:
The method returns the `RouteParentStatus` reference because later the caller needs to access the conditions field to see if it has been accepted.